src/ngraph/op/extractimagepatches.cpp

```diff
@@ -0,0 +1,106 @@
+//*****************************************************************************
+// Copyright 2017-2020 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#include "ngraph/op/extractimagepatches.hpp"
+#include "ngraph/attribute_visitor.hpp"
+
+using namespace std;
+using namespace ngraph;
+
+// ExtractImagePatches v3
+
+constexpr NodeTypeInfo op::v3::ExtractImagePatches::type_info;
+
+op::v3::ExtractImagePatches::ExtractImagePatches(const Output<Node>& image,
+                                                 const op::v3::ExtractImagePatchesAttrs& attrs)
+    : Op({image})
+    , m_attrs(attrs)
+{
+    constructor_validate_and_infer_types();
+}
+
+void op::v3::ExtractImagePatches::validate_and_infer_types()
+{
+    Shape input_shape = get_input_shape(0);
+
```

Comment:
The input shape could be dynamic, in which case you make the output shape dynamic and skip the shape-based checking.

Reply:
get_partial_shape, then, check if it is static or not to determine the next action.

Reply:
thanks I will update this